From 92936ce9fcc71e99922296b83a0617180d091e59 Mon Sep 17 00:00:00 2001
From: Luiz Augusto von Dentz <luiz.dentz-von@nokia.com>
Date: Fri, 3 Sep 2010 12:20:55 +0300
Subject: [PATCH] Fix memory leak and possible crash when removing audio device

Memory allocated for pending authorizations were not freed when
removing/unregistering an audio device, this could also cause a crash
if the core doesn't automatically cancel pending requests.
---
 audio/device.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/audio/device.c b/audio/device.c
index 155304e..b30590e 100644
--- a/audio/device.c
+++ b/audio/device.c
@@ -111,6 +111,8 @@ static void device_free(struct audio_device *dev)
 	btd_device_unref(dev->btd_dev);
 
 	if (priv) {
+		if (priv->auths)
+			audio_device_cancel_authorization(dev, NULL, NULL);
 		if (priv->control_timer)
 			g_source_remove(priv->control_timer);
 		if (priv->avdtp_timer)
-- 
1.7.0.4

